test/templates/fake_middleware.rb

```diff
@@ -0,0 +1,42 @@
+require "action_view"
+
+# Fake rack app to render templates
+class FakeMiddleware
+  def call(env)
+    [200, header, [render(req_path(env))]]
+  end
+
+  # default header
+  def header
+    {"Content-Type" => "application/javascript"}
+  end
+
+  # extract target path from REQUEST_PATH
+  def req_path(env)
+    env["REQUEST_PATH"].match(req_path_regex)[1]
+  end
+
+  def view_path
+    raise "view_path() is not implemented"
```

Comment:
There is a `NotImplemented` error for this case, but I would just use the duck type here (just skip it and let Ruby raise). We won't need a really stable interface and don't have that many clients for it so we can help confused developers. :-)